frontend/layouts/discover-page/component.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { FC, useEffect, useMemo, useState } from 'react';
+
+import cx from 'classnames';
+
+import { useRouter } from 'next/router';
+
+import DiscoverSearch from 'containers/discover-search';
+
+import LayoutContainer from 'components/layout-container';
+import { Paths } from 'enums';
+
+import { useProjectsList } from 'services/projects/projectService';
+
+import Header from './header';
+import Navigation from './navigation';
+import { DiscoverPageLayoutProps } from './types';
+
+export const DiscoverPageLayout: FC<DiscoverPageLayoutProps> = ({
+  screenHeightLg = false,
+  children,
+}: DiscoverPageLayoutProps) => {
+  const router = useRouter();
+  const { query } = router;
+
+  // We're keeping data in a state so that, while we're fetching new data,
+  // we can keep the existing one visible in the screen (and add a loading spinner).
```

Comment:
Not sure if this applies here, but you know react-query can do that for you automatically, right? https://react-query.tanstack.com/guides/paginated-queries#better-paginated-queries-with-keeppreviousdata

Reply:
It does apply and I knew about `keepPreviousData`, but for some reason I completely forgot about it 😓